gui.js

```diff
@@ -1480,6 +1481,41 @@ IDE_Morph.prototype.createCorralBar = function () {
         this.corralBar.left() + padding + newbutton.width() + padding
     );
     this.corralBar.add(paintbutton);
+
+    cambutton = new PushButtonMorph(
+        this,
+        "newCamSprite",
+        new SymbolMorph("camera", 15)
+    );
+
+    cambutton.corner = 12;
+    cambutton.color = colors[0];
+    cambutton.highlightColor = colors[1];
+    cambutton.pressColor = colors[2];
+    cambutton.labelMinExtent = new Point(36, 18);
+    cambutton.padding = 0;
+    cambutton.labelShadowOffset = new Point(-1, -1);
+    cambutton.labelShadowColor = colors[1];
+    cambutton.labelColor = this.buttonLabelColor;
+    cambutton.contrast = this.buttonContrast;
+    cambutton.drawNew();
+    cambutton.hint = "take a camera snapshot and\nimport it as a new sprite";
+    cambutton.fixLayout();
+    cambutton.setCenter(this.corralBar.center());
+    cambutton.setLeft(
+        this.corralBar.left() + padding + newbutton.width() + padding + paintbutton.width() + padding
+    );
+
+    if (location.protocol === 'http:') {
```

Comment:
Does this work OK with file:// URLs?

Also, not a blocker but just curious -- do you know if the same restriction applies to Electron / nw.js apps?

Reply:
The restriction does not apply for file:// URLs, but accessing the camera from these [won't work anyway](https://stackoverflow.com/questions/17682978/getting-not-allowed-to-load-local-resource-error-while-trying-to-attach-a-medi), it will give out this error:

`Not allowed to load local resource: blob:null/e90fa29d-f409-442a-94ba-98fa9999eda1`

This works great in nw.js apps, I'm using this very same code in Snap4Arduino :)

Reply:
Good to know about the desktop stuff. That post seems to suggest HTTP is OK with a localhost server? :O That doesn't make sense to me, but I wonder if we can support that. 

Sigh, I should just test my auto-updater app some more. Then we wouldn't have to worry about offline stuff.

Reply:
Exactly, there's no issue accessing it locally with a webserver.

Reply:
In that case, this is wrong to disable the button, because you're not going to be using https, right?

Reply:
Ouch, you're right! I always run my server with https, but you're totally right.

Now there's a problem. How do we detect if the user is running http locally?

Reply:
I was going to write that last line, and the realized there's not a good way. :/

We could just check if the location is `localhost` or `127.0.0.1` or any of the few others, but that seems annoying. 

Perhaps what we should do is just try to create a video stream, (inside a try/catch) and store whether or not it's successful? That would solve the IE disabling issue too. 

`IDE_Morph.prototype.videoCaptureAvaialbe = function () {...}`

I don't know if there's a performance hit to doing that or if we can cache the video element to be used if you do open the dialog?

Actually, this might be a better check anyway because you might be on a device without a webcam.